src/mrtc/agentManagerListener/AutoAuthorizeListener.java

```
@@ -55,4 +63,28 @@ public void agentUnregistered(@NotNull SBuildAgent sBuildAgent) {
       sBuildAgent.setAuthorized(false, null, "ASG agent unregistered");
     }
   }
+  
+  private void moveAgentToPool(@NotNull SBuildAgent sBuildAgent) {
+    Map<String,String> parameters = sBuildAgent.getAvailableParameters();
+
+    if (!parameters.containsKey("agentPool")) {
```

Should this cover `agentPool=` case? What's the expected behavior in that case?

---

The behavior I intended is that the pool join would 'fail silently' by auto-joining the default pool if the specified pool doesn't exist. It doesn't appear you can create an empty string agent pool, so this would fall into the generic case covered on line 79. In this case, a warning will be logged in the `teamcity-server.log` file to help diagnose the issue. Perhaps I could add an explicit empty string check to make this case more obvious?

---

Yeah, I was looking at the code from the perspective that it's being organized into base cases and core logic, and empty string seemed like a base case. Either way works, but it depends on if you'd want the contributors to look at it as an invalid case (from the configs perspective) or a no-op in the assignment phase.